modules/rostests/apitests/shell32/ItemIDList.cpp

```diff
@@ -0,0 +1,133 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     Test for SHSimpleIDListFromPath
+ * COPYRIGHT:   Copyright 2024 Katayama Hirofumi MZ (katayama.hirofumi.mz@gmail.com)
+ */
+
+#include "shelltest.h"
+#include <shellutils.h>
+
+enum { DIRBIT = 1, FILEBIT = 2 };
+
+static int FileStruct_Att(LPCITEMIDLIST pidl)
+{
+    return pidl && pidl->mkid.cb > 14 ? *((WORD*)((BYTE*)(pidl) + 12)) : (1 << 31); // See FileStruct in pidl.h
+}
+
+START_TEST(SHSimpleIDListFromPath)
+{
+    HRESULT hr;
+    WCHAR szPath[MAX_PATH];
+    GetWindowsDirectoryW(szPath, _countof(szPath));
+    LPITEMIDLIST item;
+
+    // We compare pidl1 and pidl2
+    CComHeapPtr<ITEMIDLIST> pidl1(SHSimpleIDListFromPath(szPath));
+    CComHeapPtr<ITEMIDLIST> pidl2(ILCreateFromPathW(szPath));
+
+    // Yes, they are equal logically
+    LPITEMIDLIST pidl1Last = ILFindLastID(pidl1), pidl2Last = ILFindLastID(pidl2);
+    ok_int(ILIsEqual(pidl1, pidl2), TRUE);
+    ok_int(ILIsEqual(pidl1Last, pidl2Last), TRUE);
+
+    // Bind to parent
+    CComPtr<IShellFolder> psf1, psf2;
+    hr = SHBindToParent(pidl1, IID_PPV_ARG(IShellFolder, &psf1), NULL);
+    ok_long(hr, S_OK);
+    hr = SHBindToParent(pidl2, IID_PPV_ARG(IShellFolder, &psf2), NULL);
+    ok_long(hr, S_OK);
+
+    // Get attributes
+    DWORD attrs1 = SFGAO_FOLDER, attrs2 = SFGAO_FOLDER;
+    hr = (psf1 ? psf1->GetAttributesOf(1, &pidl1Last, &attrs1) : E_UNEXPECTED);
+    ok_long(hr, S_OK);
+    hr = (psf2 ? psf2->GetAttributesOf(1, &pidl2Last, &attrs2) : E_UNEXPECTED);
+    ok_long(hr, S_OK);
+
+    // There is the difference in attributes because SHSimpleIDListFromPath
+    // cannot create PIDLs to folders, only files and drives:
+    ok_long((attrs1 & SFGAO_FOLDER), 0);
+    ok_long((attrs2 & SFGAO_FOLDER), SFGAO_FOLDER);
+
+
+    // Make sure the internal details match Windows NT5+
+    GetSystemDirectoryW(szPath, _countof(szPath));
+    CComHeapPtr<ITEMIDLIST> pidlSys32(SHSimpleIDListFromPath(szPath));
+    if (szPath[1] != ':' || PathFindFileNameW(szPath) <= szPath)
+        skip("Not a local directory %ls\n", szPath);
+    else if (!(LPITEMIDLIST)pidlSys32)
+        skip("?\n");
```

Comment:
Same comment for all the other if/else(if) blocks below: https://reactos.org/wiki/Coding_Style#Braces